ci-operator/step-registry/ibmcloud/provision/vpc/ibmcloud-provision-vpc-commands.sh

```diff
@@ -0,0 +1,201 @@
+    local intput_string="$1" seperator="$2" output_array
+
+    if [[ -z "$seperator" ]]; then
+        seperator=","
+    fi
+
+    IFS="$seperator" read -r -a output_array <<< "${intput_string}"
+    echo "${output_array[@]}"
+}
+
+function getAddressPre() {
+    local vpc_info_file="$1"
+    local ip ips
+    ip=$(cat ${vpc_info_file} | jq -c -r .address_prefixes[0].cidr)
+    IFS="." read -ra ips <<< "${ip}"
+    echo "${ips[0]}.${ips[1]}.0.0/16"
+}
+
+function create_zone_public_gateway() {
+    local gateName="$1" vpcName="$2" zone="$3"
+
+    "${IBMCLOUD_CLI}" is public-gateway-create ${gateName} ${vpcName} ${zone} || return 1
+}
+
+function attach_public_gateway_to_subnet() {
+    local subnetName="$1" vpcName="$2" pgwName="$3"
+
+    "${IBMCLOUD_CLI}" is subnet-update ${subnetName} --vpc ${vpcName} --pgw ${pgwName} || return 1
+}
+
+checkCli
+
+
+ibmcloud_login || exit 1
+
+## Create the VPC
+echo "$(date -u --rfc-3339=seconds) - Creating the VPC..."
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+region="${LEASED_RESOURCE}"
+
+resource_group="${CLUSTER_NAME}-rg"
+
+create_resource_group ${resource_group}
+echo "${resource_group}" > "${SHARED_DIR}/ibmcloud_resource_group"
+
+echo "Create VPC..."
+vpc_name="${CLUSTER_NAME}-vpc"
+echo "${vpc_name}" > "${SHARED_DIR}/ibmcloud_vpc_name"
+create_vpc "${CLUSTER_NAME}" "${vpc_name}" "${resource_group}" "${region}" || exit 3
+
+vpc_info_file="${ARTIFACT_DIR}/vpc_info"
+check_vpc "${vpc_name}" "${vpc_info_file}" || exit 3
+
+controlPlaneSubnets=$(cat "${vpc_info_file}" | jq -c -r '[.subnets[] | select(.name|test("control-plane")) | .name]')
+IFS="," read -r -a controlPlaneSubnetAry <<< ${controlPlaneSubnets:1:-1}
+computeSubnets=$(cat "${vpc_info_file}" | jq -c -r '[.subnets[] | select(.name|test("compute")) | .name]')
+IFS="," read -r -a computeSubnetsAry <<< ${computeSubnets:1:-1}
+vpcAddressPre=$(getAddressPre ${vpc_info_file})
+
+if [[ "${RESTRICTED_NETWORK}" = "yes" ]]; then
+    echo "[WARN] Skip creating public gateway to create disconnected network"
+else
+    zones=("${region}-1" "${region}-2" "${region}-3")
+    for zone in "${zones[@]}"; do
+        echo "Creating public gateway in ${zone}..."
+        public_gateway_name="${CLUSTER_NAME}-gateway-${zone}"
+        create_zone_public_gateway "${public_gateway_name}" "${vpc_name}" "$zone" || exit 1
+        for subnet in $(cat "${vpc_info_file}" | jq -r --arg z "${zone}" '.subnets[] | select(.zone.name==$z) | .name'); do
+            echo "Attaching public gateway - ${public_gateway_name} to subnet - ${subnet}..."
+            attach_public_gateway_to_subnet "${subnet}" "${vpc_name}" "${public_gateway_name}" || exit 1
+        done
+    done
+fi
+
+cat > "${SHARED_DIR}/customer_vpc_subnets.yaml" << EOF
+platform:
+  ibmcloud:
+    resourceGroupName: ${resource_group}
+    vpcName: ${vpc_name}
+    controlPlaneSubnets:
+    - ${controlPlaneSubnetAry[0]}
+    - ${controlPlaneSubnetAry[1]}
+    - ${controlPlaneSubnetAry[2]}
```

Comment:
The format of `controlPlaneSubnets`'s value looks like `["knarra1205-control-plane-us-east-3", "knarra1205-control-plane-us-east-2", "knarra1205-control-plane-us-east-1"]`, it can be directly referenced in `controlPlaneSubnets: ${controlPlaneSubnets}`. 


If you really care about pretty print, you can:
```
cat "${vpc_info_file}" | jq -c -r '[.subnets[] | select(.name|test("control-plane")) | .name]' | yq-go r -P - >controlPlaneSubnets.yaml
yq w -i "${SHARED_DIR}/customer_vpc_subnets.yaml" 'platform.ibmcloud.controlPlaneSubnets' -f controlPlaneSubnets.yaml
```